src/shim/umq/hwq.cpp

```diff
@@ -1,24 +1,31 @@
 // SPDX-License-Identifier: Apache-2.0
 // Copyright (C) 2023-2025, Advanced Micro Devices, Inc. All rights reserved.
 
-#include <x86intrin.h>
 #include "bo.h"
 #include "hwq.h"
+#if defined(__x86_64__) || defined(_M_X64)
+  #include <x86intrin.h>
+#endif
 
 namespace {
 
-// flash cache line for non coherence memory
-inline void
-clflush_data(void *data, int len)
-{
-	const int LINESIZE = 64;
-	const char *cur = (const char *)data;
-	// must be at least one cache line
-	uintptr_t lastline = (uintptr_t)(cur + len - 1) | (LINESIZE - 1);
-	do {
-		_mm_clflush(cur);
-		cur += LINESIZE;
-	} while (cur <= (const char *)lastline);
+void flush_cache_line(const char *cur) {
+  const int LINESIZE = 64;
+#if defined(__x86_64__) || defined(_M_X64)
+  _mm_clflush(cur);
+#else
+  __builtin___clear_cache((char *)cur, (char *)(cur + LINESIZE));
+#endif
+}
+
+void clflush_data(const void *data, size_t len) {
+  const int LINESIZE = 64;
```

Comment:
LINESIZE is also used in flush_cache_lin() and are both 64. Use macro is better.